pkg/controller/applicationlayer/applicationlayer_controller_test.go

```diff
@@ -0,0 +1,152 @@
+			mockStatus.On("AddDeployments", mock.Anything).Return()
+			mockStatus.On("IsAvailable").Return(true)
+			mockStatus.On("AddStatefulSets", mock.Anything).Return()
+			mockStatus.On("AddCronJobs", mock.Anything)
+			mockStatus.On("OnCRFound").Return()
+			mockStatus.On("ClearDegraded")
+			mockStatus.On("SetDegraded", "Waiting for LicenseKeyAPI to be ready", "").Return().Maybe()
+			mockStatus.On("ReadyToMonitor")
+
+			r = ReconcileApplicationLayer{
+				client:          c,
+				scheme:          scheme,
+				provider:        operatorv1.ProviderNone,
+				status:          mockStatus,
+				licenseAPIReady: &utils.ReadyFlag{},
+			}
+
+			Expect(c.Create(ctx, &operatorv1.Installation{
+				ObjectMeta: metav1.ObjectMeta{Name: "default"},
+				Spec: operatorv1.InstallationSpec{
+					Variant:  operatorv1.TigeraSecureEnterprise,
+					Registry: "some.registry.org/",
+				},
+				Status: operatorv1.InstallationStatus{
+					Variant: operatorv1.TigeraSecureEnterprise,
+					Computed: &operatorv1.InstallationSpec{
+						Registry: "my-reg",
+						// The test is provider agnostic.
+						KubernetesProvider: operatorv1.ProviderNone,
+					},
+				},
+			})).NotTo(HaveOccurred())
+
+			Expect(c.Create(ctx, &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+				Spec: crdv1.FelixConfigurationSpec{
+					TPROXYMode: nil,
+				},
+			})).NotTo(HaveOccurred())
+
+			// Mark that the watch for license key was successful.
+			r.licenseAPIReady.MarkAsReady()
+		})
+
+		It("should render accurate resources for for log collection", func() {
+
+			enabled := operatorv1.L7LogCollectionEnabled
+			// Apply the logcollector CR to the fake cluster.
+			Expect(c.Create(ctx, &operatorv1.ApplicationLayer{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ApplicationLayerSpec{
+					LogCollection: &operatorv1.LogCollectionSpec{
+						CollectLogs: &enabled,
+					},
+				},
+			})).NotTo(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			ds := appsv1.DaemonSet{
+				TypeMeta: metav1.TypeMeta{Kind: "DaemonSet", APIVersion: "apps/v1"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      applicationlayer.L7LogCollectorDeamonsetName,
+					Namespace: common.CalicoNamespace,
+				},
+			}
+
+			Expect(test.GetResource(c, &ds)).To(BeNil())
+			Expect(ds.Spec.Template.Spec.Containers).To(HaveLen(2))
+
+			proxy := ds.Spec.Template.Spec.Containers[0]
+			Expect(proxy).ToNot(BeNil())
+			Expect(proxy.Image).To(Equal(fmt.Sprintf("some.registry.org/%s:%s",
+				components.ComponentEnvoyProxy.Image, components.ComponentEnvoyProxy.Version)))
+
+			l7collector := ds.Spec.Template.Spec.Containers[1]
+			Expect(l7collector).ToNot(BeNil())
+			Expect(l7collector.Image).To(Equal(fmt.Sprintf("some.registry.org/%s:%s",
+				components.ComponentL7Collector.Image, components.ComponentL7Collector.Version)))
+		})
+
+	})
```

Comment:
Please include a Reconcile test that with no LogCollection resource the tproxyMode field will be disabled.

Reply:
added tests an improved logging here 1a8e3f97e51178ba8e5c27e6c82fa6cc5d5b975d